src/polyfills/intersection-observer.js

```diff
@@ -0,0 +1,197 @@
+
+/**
+ * @fileoverview
+ * See https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver.
+ */
+
+
+/* DO NOT SUBMIT: example:
+var inob = new IntersectionObserver(function(records) {
+  console.log('inob: ', records);
+});
+inob.observe($0);
+*/
+
+
+/**
+ * @param {!Window} win
+ */
+export function install(win) {
+  // DO NOT SUBMIT: remove, testing.
+  win.QqqqInOb = win.IntersectionObserver;
+  win.QqqqInObEntry = win.IntersectionObserverEntry;
+  delete win.IntersectionObserver;
+  delete win.IntersectionObserverEntry;
+
+  if (!win.IntersectionObserver) {
+    win.IntersectionObserver = /** @type {typeof IntersectionObserver} */ (IntersectionObserverStub);
+  }
+}
+
+/**
+ * @param {!Window} win
+ */
+export function shouldLoadPolyfill(win) {
+  return (!win.IntersectionObserver || win.IntersectionObserver === IntersectionObserverStub);
+}
+
+/**
+ * @param {!Window} win
+ * @param {function()} installer
+ */
+export function upgradePolyfill(win, installer) {
+  // Can't use the IntersectionObserverStub here directly since it's a separate
+  // instance deployed in v0.js vs the polyfill extension.
+  const Stub = /** @type {typeof IntersectionObserverStub} */ (win.IntersectionObserver);
+  if (!Stub || Stub.stubs_) {
```

Comment:
I think this logic is broken. Seems we should be testing for `Stubs` (and if it's not, then something is seriously broken) only?

Reply:
Do you mean `!Stub` is not needed here?

Reply:
I think the boolean logic got reversed. We should only run this if-statement's body if `Stubs && Stubs.stubs`, right?

Reply:
This way the polyfill will install even if somehow stub wasn't set up. I can reverse as you suggest, but this way seems to cover more edge cases.

Reply:
Right. Confused myself on this one. I redid this and commented it heavily. I think it should be clear now. There are two modes there: complete polyfill (stub -> polyfill) and partial polyfill when only some features were added, such as `isIntersecting`.